src/vs/platform/notification/common/notification.ts

```diff
@@ -62,6 +62,12 @@ export interface INotification extends INotificationProperties {
 	 * this usecase and much easier to use!
 	 */
 	actions?: INotificationActions;
+
+	/**
+	 * options for showing "never show again" prompt.
+	 * by default shown as secondary action
+	 */
+	neverShowOptions?: INeverShowOptions;
```

Comment:
Move this into `INotificationProperties` you do not need it both here and in `IPromptOptions`

Reply:
Where is best to document the different default behaviors?

Reply:
What are the differences?

Reply:
The default in the prompt method is to show the "never show again" as primary action, while in the notify method the default is secondary (I thought that's what you [suggested](https://github.com/microsoft/vscode/pull/73968#issuecomment-498664661). Or maybe you meant secondary in both?)  

the different behavior is documented separately in each interface.

Reply:
I think it is OK that the option is implemented differently depending on where it is being used.